Parlance.Project/TranslationFiles/AppleStrings/AppleStringsTranslationFile.cs

```diff
@@ -0,0 +1,119 @@
+using System.Security.Cryptography;
+using System.Text;
+using Parlance.CldrData;
+using Parlance.Project.Index;
+
+namespace Parlance.Project.TranslationFiles.AppleStrings;
+
+[TranslationFileType("apple-strings", ExpectedTranslationFileNameFormat.Underscored)]
+public class AppleStringsTranslationFile : ParlanceTranslationFile, IParlanceMonoTranslationFile
+{
+        private string _baseFile = null!;
+    private Locale _baseLocale = null!;
+    private string _file = null!;
+    private Locale _locale = null!;
+
+    public AppleStringsTranslationFile(IParlanceSubprojectLanguage? subprojectLanguage,
+        IParlanceIndexingService? indexingService) : base(subprojectLanguage, indexingService)
+    {
+    }
+
+    public override string Hash { get; internal set; } = null!;
+    public override IList<IParlanceTranslationFileEntry> Entries { get; internal set; } = null!;
+
+    public static async Task<ParlanceTranslationFile> CreateAsync(string file, Locale locale, string baseFile,
+        Locale baseLocale,
+        IParlanceSubprojectLanguage? subprojectLanguage, IParlanceIndexingService? indexingService)
+    {
+        var translationFile = new AppleStringsTranslationFile(subprojectLanguage, indexingService);
+        await translationFile.LoadFile(file, locale, baseFile, baseLocale);
+        return translationFile;
+    }
+
+    private string UnescapeString(string str)
+    {
+        if (str.StartsWith("\"") && str.EndsWith("\""))
+        {
+            str = str[1..^1];
+        }
+        
+        return str.Replace("\\r", "\r").Replace("\\n", "\n").Replace("\\\"", "\"").Replace("\\\\", "\\");
+    }
+
+    private string EscapeString(string str)
+    {
+        return $"\"{str.Replace("\\", "\\\\").Replace("\n", "\\n").Replace("\r", "\\r").Replace("\"", "\\\"")}\"";
+    }
+
+    private async Task LoadFile(string file, Locale locale, string baseFile, Locale baseLocale)
+    {
+        _file = file;
+        _locale = locale;
+        _baseFile = baseFile;
+        _baseLocale = baseLocale;
+
+        var fileContents = await File.ReadAllTextAsync(file);
+        var baseFileContents = await File.ReadAllLinesAsync(baseFile);
+        Hash = Convert.ToHexString(SHA256.HashData(Encoding.UTF8.GetBytes(fileContents)));
+
+        var baseDocKeys = baseFileContents.Select(x =>
+        {
+            if (!x.EndsWith(";")) throw new("No trailing ; in Apple Strings file");
+            var splitter = x.IndexOf("=", StringComparison.Ordinal);
+            return (UnescapeString(x[..splitter].Trim()), UnescapeString(x[(splitter + 1)..^1].Trim()));
```

Comment:
```suggestion
            var split = x.Split('=', StringSplitOptions.TrimEntries);
            Debug.Assert(split.Length == 2);
            return (Key: UnescapeString(split[0]), Source: UnescapeString(split[1][..^1].TrimEnd()));
```

This doesn't handle the value having an `=` character in it, but neither does the original code.